### ISSUE TITLE
AssertionError when generating docs

### ISSUE THREAD

**Issue by anderslanglands** [mosra/m.css#33]
Hi, I love the look of this project. Unfortunately it's failing for me on my first try. I'm following the instructions on the website and running:
`./dox2html5.py ../../../build/Doxyfile-mcss`

I get the following output:
```
Generate XML output for dir blahblahblah... (lots of times)
lookup cache used 1885/65536 hits=14417 misses=2047
finished...
Traceback (most recent call last):
  File "./dox2html5.py", line 2862, in <module>
    run(doxyfile, os.path.abspath(args.templates), args.wildcard, args.index_pages, search_merge_subtrees=not args.search_no_subtree_merging, search_add_lookahead_barriers=not args.search_no_lookahead_barriers, search_merge_prefixes=not args.search_no_prefix_merging)
  File "./dox2html5.py", line 2780, in run
    parsed = parse_xml(state, file)
  File "./dox2html5.py", line 2308, in parse_xml
    func = parse_func(state, memberdef)
  File "./dox2html5.py", line 1462, in parse_func
    func.description, templates, params, func.return_value, func.return_values, search_keywords, func.is_deprecated = parse_func_desc(state, element)
  File "./dox2html5.py", line 1310, in parse_func_desc
    parsed = parse_desc_internal(state, element.find('detaileddescription'))
  File "./dox2html5.py", line 698, in parse_desc_internal
    add_css_class=add_css_class)
  File "./dox2html5.py", line 759, in parse_desc_internal
    out.parsed += '<li>{}</li>'.format(parse_desc(state, li))
  File "./dox2html5.py", line 1256, in parse_desc
    assert not parsed.templates and not parsed.params and not parsed.return_value
AssertionError

```

Any idea how to go about debugging further?

**Comment by anderslanglands** [comment]
Hmm seems to be something in my config. If I generate a default Doxyfile it works. Only trouble now is the output is very broken. I have a giant magnifying glass and no styling on any of the elements. If I inspect the html I can't see any reference to css.

**Comment by mosra** [comment]
Hi, thanks for the report!

If you run `dox2html5.py` with the `--debug` flag, it will show which particular XML file it is processing before it dies. In that file there seems to be something that it doesn't expect. Can you upload that particular file somewhere so I can check it?

Regarding the missing stylesheets: if it crashes before it can finish, it will not copy the stylesheets to the output, resulting in what you describe (and yes, the giant magnifying glass is terrible, i have to fix that as well). The CSS files should be referenced directly in the `<head>`, unless you have `HTML_EXTRA_STYLESHEET` overriden.

**Comment by anderslanglands** [comment]
[group__nsi.txt](https://github.com/mosra/m.css/files/1889067/group__nsi.txt)

I've uploaded the last file it said it was parsing (renamed it from .xml to .txt to get github to upload it)

**Comment by anderslanglands** [comment]
I get the missing stylesheets when I'm just using a freshly generated (default) Doxyfile, for which dox2html5.py appears to succeed. The `<head>` of the generated output looks like this:
```
<head>
  <meta charset="UTF-8" />
  <title>My Project</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <meta name="theme-color" content="#22272e" />
</head>
```

**Comment by anderslanglands** [comment]
I've sent you an email with the docstring and doxyfile.

I can confirm that https://github.com/mosra/m.css/commit/90c0c902f3dceaddff0b4c613178a47617db26e4 fixes the stylesheet issue I was having with the default Doxyfile, but now I can't see any documentation from the main pages (i.e. when I click on Modules, Classes and Files from the index, they are all empty). I can find documentation for a particular class using the search feature (which is great btw), but nothing is added to the main pages.

**Comment by mosra** [comment]
> when I click on Modules, Classes and Files from the index, they are all empty

This might be due to the following [feature](http://mcss.mosra.cz/doxygen/#important-differences-to-stock-html-output):

> - Files, di­rec­to­ries and sym­bols that don’t have any doc­u­men­ta­tion are not present in the out­put at all. This is done in or­der to en­cour­age good doc­u­men­ta­tion prac­tices.

In particular, when a directory, namespace or a file doesn't have any documentation (no documentation block, not even a `@brief`), neither it *nor its child items* are present in the index tree. In your case I'd check the `hikari/` folder if it has some `@dir` docs and the `hikari` and `hikari::nsi` namespaces if they have corresponding doc blocks.

At the moment it's hardcoded like this (in order to force *me* to write proper docs for my projects) and you might have a different opinion -- if you complain loudly enough, I could make this feature opt-out  ;) (but I don't think it's *too bad*...)

**Comment by anderslanglands** [comment]
Ahhh ok that’s probably it. I don’t think it’s a bad idea for the reasons
you state, however maybe it would be a good idea to print a warning summary
at the end of the process to say “the documentation for classes X, Y and Z”
was ignored because the enclosing namespace was not documented”. It wasn’t
immediately obvious to me from reading the documentation on your site that
this would be the expected behaviour.

**Comment by mosra** [comment]
It probably maybe says that somewhere buried deep in the thousands of lines of the `--debug` output, but definitely not clearly enough. I'll improve that.

**Comment by mosra** [comment]
The core problem for this assertion error was fixed in e8e0f60e721f5cf582fa4885010648f620fdea53. I tried to improve the warning output for ignored docs, but it's giving a lot of false positives and I didn't come up with any robust enough solution for that. So the only option is to have a look at the `--debug` output. Maybe I come up with something later.
